kube_apiserver_metrics/datadog_checks/kube_apiserver_metrics/kube_apiserver_metrics.py

```diff
@@ -0,0 +1,134 @@
+        }
+        self.bearer_token_found = False
+        # Create instances we can use in OpenMetricsBaseCheck
+        generic_instances = None
+        if instances is not None:
+            generic_instances = self.create_generic_instances(instances)
+
+        super(KubeApiserverMetricsCheck, self).__init__(
+            name,
+            init_config,
+            agentConfig,
+            instances=generic_instances,
+            default_instances={
+                "kube_apiserver": {
+                    'namespace': 'kube_apiserver',
+                    'metrics': [
+                        {
+                            'apiserver_current_inflight_requests': 'current_inflight_requests',
+                            'apiserver_longrunning_gauge': 'longrunning_gauge',
+                            'go_threads': 'go_threads',
+                            'go_goroutines': 'go_goroutines',
+                            'APIServiceRegistrationController_depth': 'APIServiceRegistrationController_depth',
+                            'etcd_object_counts': 'etcd_object_counts',
+                        }
+                    ],
+                }
+            },
+            default_namespace="kube_apiserver",
+            )
+
+    def check(self, instance):
+        if not self.bearer_token_found:
+            self.log.warn("Not using the service account bearer token file, AuthN/Z will fail.")
+        if not instance.get("prometheus_url"):
+            raise ConfigurationError("Missing prometheus_endpoint field. Skipping check")
+        scraper_config = self.get_scraper_config(instance)
+        self.process(scraper_config, metric_transformers=self.METRIC_TRANSFORMERS)
+
+    def create_generic_instances(self, instances):
+        """
+        Transform each Kubernetes APIServer endpoint instance into a OpenMetricsBaseCheck instance
+        """
+        generic_instances = []
+        for instance in instances:
+            transformed_instance = self._create_kube_apiserver_metrics_instance(instance)
+            generic_instances.append(transformed_instance)
+
+        return generic_instances
+
+    def _create_kube_apiserver_metrics_instance(self, instance):
+        """
+        Set up kube_apiserver_metrics instance so it can be used in OpenMetricsBaseCheck
+        """
+        kube_apiserver_metrics_instance = deepcopy(instance)
+        endpoint = instance.get('prometheus_url')
+        scheme = instance.get('scheme', self.DEFAULT_SCHEME)
+
+        kube_apiserver_metrics_instance['prometheus_url'] = "{0}://{1}".format(scheme, endpoint)
+
+        bearer_token_path = instance.get('bearer_token_path', self.DEFAULT_BEARER_TOKEN_PATH)
+        bearer_token = ""
+        if not os.path.isfile(bearer_token_path) or not os.access(bearer_token_path, os.R_OK):
+            self.bearer_token_found = False
+        else:
+            with open(bearer_token_path, "r") as f:
+                bearer_token = f.read()
```

Comment:
Let's just try this and ignore errors

Reply:
I'd still want to log a warning if the bearer token can't be read or if the file does not exist.
What do you think of removing the if logic above and use:
```
        try:
            with open(bearer_token_path, "r") as f:
                bearer_token = f.read()
        except Exception as err:
            self.log.warning("Could not retrieve the bearer token file: {}".format(err))
```